File: bindings/java/src/main/java/com/keynub/licdongle/Scope.java
package com.keynub.licdongle;

/** Scope for app-data envelope encryption ({@link Session#appEncrypt}). */
public enum Scope {
    /** Only this physical dongle can decrypt (node-locking). */
    DEVICE(0),
    /** Any dongle from the same developer batch can decrypt. */
    DEVELOPER(1);

    private final int code;

    Scope(int code) {
        this.code = code;
    }

    /** The numeric scope code passed to the native layer. */
    public int code() {
        return code;
    }
}

File: bindings/matlab/+keynub/Scope.m
classdef Scope < uint8
%KEYNUB.SCOPE Who can decrypt data produced by keynub.Session.appEncrypt.
%
%   Device    - only this one physical dongle can decrypt it.
%   Developer - any dongle issued by the same developer can decrypt it, which
%               is what you want for data shipped to every customer.
%
%   See also keynub.Session/appEncrypt.

    enumeration
        Device    (0)
        Developer (1)
    end
end
